<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Figure 5: gene pathway; triple mutants; TMB adjusted vs multi-cancer model</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genie_ras_hallmarks</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Figure 5: gene pathway; triple mutants; TMB
adjusted vs multi-cancer model</h1>
<h4 class="date">18 August, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>genie/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220803code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220803)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220803code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220803)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong74b2b93">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 74b2b93
</a>
</p>
</div>
<div id="strongRepositoryversionstrong74b2b93"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 74b2b93.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  analysis/co-mutation.Rmd
    Untracked:  analysis/fig_s1.Rmd
    Untracked:  analysis/fig_s10.Rmd
    Untracked:  analysis/fig_s11.Rmd
    Untracked:  analysis/fig_s12.Rmd
    Untracked:  analysis/fig_s13.Rmd
    Untracked:  analysis/fig_s14.Rmd
    Untracked:  analysis/fig_s15.Rmd
    Untracked:  analysis/fig_s16.Rmd
    Untracked:  analysis/fig_s17.Rmd
    Untracked:  analysis/fig_s18.Rmd
    Untracked:  analysis/fig_s19.Rmd
    Untracked:  analysis/fig_s2.Rmd
    Untracked:  analysis/fig_s20.Rmd
    Untracked:  analysis/fig_s3.Rmd
    Untracked:  analysis/fig_s4.Rmd
    Untracked:  analysis/fig_s5.Rmd
    Untracked:  analysis/fig_s6.Rmd
    Untracked:  analysis/fig_s7.Rmd
    Untracked:  analysis/fig_s8.Rmd
    Untracked:  analysis/fig_s9.Rmd
    Untracked:  analysis/methods_comparison.Rmd
    Untracked:  analysis/potential_confounding.Rmd
    Untracked:  code/TCGA-COAD/
    Untracked:  code/TCGA-LUAD-trips/
    Untracked:  code/TCGA-LUAD/
    Untracked:  code/TCGA-PAAD/
    Untracked:  code/co-mutation/
    Untracked:  code/misc/
    Untracked:  code/sets/
    Untracked:  code/stan/
    Untracked:  output/co-mutation/
    Untracked:  output/gsea_data_wrangling.Rmd/
    Untracked:  output/methods-comparison/
    Untracked:  output/mutsig_s13/
    Untracked:  output/prevalence/
    Untracked:  public/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitattributes
    Modified:   .gitignore
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/fig7.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Modified:   code/README.md
    Modified:   data/README.md
    Modified:   genie_ras_hallmarks.Rproj
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/fig5.Rmd</code>) and HTML
(<code>docs/fig5.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
af43f9e
</td>
<td>
abalan2
</td>
<td>
2022-08-04
</td>
<td>
add code for fig5
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="panel-a-gene-pathway-results" class="section level1">
<h1>Panel A: Gene pathway results</h1>
<pre class="r"><code>library(ggplot2)
library(plyr)
library(reshape2)
library(cowplot)
library(tidyverse)
library(grid)
library(gridExtra)
library(RColorBrewer)
library(readxl)
library(scales)
library(here)
library(ggrepel)
library(purrr)
library(rasfuns)</code></pre>
<pre class="r"><code>here &lt;- here::here
# Read any mutations csv for gene pathways
comut_gp &lt;- here(&quot;public&quot;,
                 &quot;table&quot;,
                 &quot;co_mutation_allele.Rmd&quot;,
                 &quot;any_mutation_(gene_pathways).csv&quot;) %&gt;%
    read_csv(show_col_types=FALSE)
# list of co-muts and cancer types to plot
plot_gp &lt;- read_excel( here( &quot;data&quot;,&quot;ras_pathway_pval.xlsx&quot;))
comut_gp &lt;-  comut_gp %&gt;%
    mutate(ras = gsub(&quot;_1&quot;, &quot;_G1&quot;,ras ))%&gt;%
    mutate(ras = gsub(&quot;_6&quot;, &quot;_Q6&quot;,ras ))%&gt;%
    mutate(index = paste0(cancer, &quot;~&quot;,
                          ras, &quot;~&quot;, nonras))
gene_pathway.levels &lt;- table(plot_gp %&gt;% select(gene_pathway)) %&gt;%
    as.data.frame() %&gt;%
    arrange(desc(Freq))
gene_pathway.levels &lt;- as.character(gene_pathway.levels$Var1)
plot.df &lt;- plot_gp %&gt;%
    mutate(ras = gsub(&quot;:&quot;, &quot;_&quot;,ras ))%&gt;%
    mutate(index = paste0(cancer_full, &quot;~&quot;, ras, &quot;~&quot;, gene_pathway))%&gt;%
    merge(., comut_gp %&gt;% select(index, p, `50%`), by.x = &quot;index&quot;, by.y = &quot;index&quot; , all.x = TRUE)%&gt;%
    mutate(cancer = factor(cancer, levels = unique(cancer))) %&gt;%
    mutate(gene_pathway = factor(gene_pathway, levels = gene_pathway.levels)) %&gt;%
    mutate(ras = factor(ras, levels = unique(ras))) %&gt;% mutate(y=0)%&gt;%
    mutate(ras_cancer = paste0(cancer, &#39;~&#39;, ras))%&gt;%
    mutate(ras_cancer = factor(ras_cancer, levels = unique(ras_cancer))) %&gt;%
    mutate(comut_type = ifelse(`50%`&gt;0, &quot;co-occurence&quot; , &quot;mutual-exclusivity&quot;))%&gt;%
    mutate(comut_type =factor(comut_type, levels = unique(comut_type)) )%&gt;%
    select(-c(&quot;logp&quot;, &quot;type&quot;))</code></pre>
<pre class="r"><code>col.pal &lt;- c(&quot;#0066cc&quot;,&quot;#cc0000&quot;)
names(col.pal) &lt;- c( &#39;Mutual\nexclusivity\n&#39;, &#39;\nCo-occurence\n&#39;)
dat.list &lt;- plot.df %&gt;%
    mutate(gene_pathway=case_when(gene_pathway==&quot;RAS_RAF_MAPK_brief&quot;~&quot;RAS_RAF_MAPK&quot;,
                                  gene_pathway==&quot;RAS_RAF_MAPK_extended&quot;~&quot;RAS_RAF_MAPK&quot;,
                                  TRUE~as.character(gene_pathway)),
           comut_type=case_when(comut_type==&quot;mutual-exclusivity&quot;~&quot;Mutual\nexclusivity\n&quot;,
                                comut_type==&quot;co-occurrence&quot;~&quot;\nCo-occurrence\n&quot;)) %&gt;%
    group_by(gene_pathway) %&gt;%
    nest()
pathways &lt;- tibble(label=c(&quot;RAS/RAF/MAPK pathway&quot;,
                           &quot;NRF2 pathway/Oxidative stress response&quot;,
                           &quot;PI3K/AKT pathway&quot;,
                           &quot;Cell cycle progression&quot;,
                           &quot;Chromatin regulation&quot;,
                           &quot;DNA damage response&quot;),
                   gene_pathway=dat.list$gene_pathway[c(3, 2, 1, 6, 4, 5)]) %&gt;%
    mutate(label=factor(label, label))
dat.list2  &lt;- dat.list %&gt;%
    left_join(pathways, by=&quot;gene_pathway&quot;) %&gt;%
    arrange(label)
lollipop_plot &lt;- function(dat){
    dat %&gt;%
        ggplot(aes(ras_cancer, p,
                   fill = comut_type,
                   color = comut_type)) +
        geom_point(aes(fill=comut_type))+
        geom_linerange(aes(ymin = y, ymax = p,
                           color = comut_type))+
        coord_flip()+
        scale_fill_manual(values = col.pal, name = &#39;&#39;,
                          guide = guide_legend(position = &#39;none&#39;,
                                               direction = &quot;vertical&quot;)) +
        scale_color_manual(values = col.pal,
                           name = &#39;&#39;,
                           guide = guide_legend(position = &#39;none&#39;,
                                                direction = &quot;vertical&quot;)) +
        ylab(&quot;-log10 p&quot;)+
        xlab(&quot;&quot;)+
        theme_classic(base_size=15) +
        theme(axis.line=element_blank(),
              axis.ticks.x = element_blank(),
              plot.margin = margin(0,0,0,0, &quot;cm&quot;),
              axis.text.y = element_text(size=13, angle = 0),
              ##axis.text.x = element_text(size=10, angle =0),
              axis.text.x=element_blank(),
              axis.title.x=element_blank(),
              panel.grid= element_blank() ,
              panel.border = element_rect(fill=NA),
              plot.title = element_text(hjust=0.5, size = 10)) +
        ylim(c(0, 9)) +
        guides(color=&quot;none&quot;, fill=&quot;none&quot;)
}
gl &lt;- map(dat.list2$data, lollipop_plot)
names(gl) &lt;- dat.list2$label
for(i in 1:6){
    gl[[i]] &lt;- gl[[i]] + ggtitle(names(gl)[i]) +
        theme(plot.title=element_text(size=16, hjust=0),
              plot.margin=unit(c(1.5, 1.5, 0, 1.5), &quot;lines&quot;))
}
gl[[6]]  &lt;- gl[[6]] +
    theme(axis.text.x=element_text(size=13),
          axis.title.x=element_text(size=17)) +
    ylab(expression(-log[10](p)))
tmp &lt;- gl[[6]] +
    theme(legend.text=element_text(size=17)) +
    guides(color=guide_legend(title=&quot;&quot;))
leg &lt;- cowplot::get_legend(tmp)
gl &lt;- map(gl, ggplotGrob)
w &lt;- gl[[1]]$widths
for(i in 2:6) gl[[i]]$widths &lt;- w
gl[[7]] &lt;- leg
heights &lt;- map_int(dat.list2$data, nrow)
heights[heights==1] &lt;- 1.5
## compensate for x-axis and title
heights[6] &lt;- 2.2
grid.arrange(grobs=gl,
             widths=c(1, 0.22),
             heights=heights,
             layout_matrix=cbind(1:6, 7))</code></pre>
<p><img src="figure/fig5.Rmd/fig5a-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>A  &lt;- arrangeGrob(grobs=gl,
                  widths=c(1, 0.22),
                  heights=heights,
                  layout_matrix=cbind(1:6, 7))
## todo
## add title to each panel
## use classic theme</code></pre>
</div>
<div id="panel-b-triple-mutations" class="section level1">
<h1>Panel B: Triple mutations</h1>
<pre class="r"><code>load(here(&quot;output&quot;, &quot;co-mutation&quot;,&quot;kras12c_tripleMuts.rda&quot;))
threshold &lt;- 1
plot.df &lt;- c()
for(cancer in names(kras12c_tripleMuts)){
    tmp.df &lt;- kras12c_tripleMuts [[cancer]] %&gt;%
        filter(intersect.percent &gt;= threshold)%&gt;%
        mutate(cancer_type = cancer)
    plot.df = rbind(plot.df ,tmp.df )
}
plot.df  &lt;-  plot.df %&gt;%
    mutate(cancer_type = recode(cancer_type, &quot;non-small cell lung cancer&quot;=&quot;NSCLC&quot; ,
                                &quot;cancer of unknown primary&quot;  = &quot;CUP&quot;,
                                &quot;colorectal cancer&quot; = &quot;CRC&quot;,
                                &quot;pancreatic cancer&quot; = &quot;PAC&quot;,
                                &quot;uterine cancer&quot; = &quot;Uterine&quot;)) %&gt;%
    mutate(gene = paste0(&quot;KRAS_&quot;, gene)) %&gt;%
    arrange(desc(intersect.percent))
cancer.levels &lt;- c(&quot;CUP&quot;, &quot;CRC&quot; ,
                   &quot;PAC&quot;, &quot;Uterine&quot;,
                   &quot;NSCLC&quot;)
plot.df$cancer_type &lt;- factor(plot.df$cancer_type, levels = cancer.levels)

gene.levels &lt;- data.frame(table(plot.df$gene)) %&gt;% arrange(desc(Freq))
gene.levels &lt;- gene.levels$Var1
plot.df &lt;- plot.df %&gt;%
    arrange(desc(intersect.percent))  %&gt;%
    arrange(factor(gene, levels = gene.levels))
tmp.df &lt;- plot.df[-c(1:4), ] %&gt;%
    arrange(desc(intersect.percent))  %&gt;%
    arrange(factor(cancer_type, levels = cancer.levels))
plot.df &lt;- rbind(plot.df[c(1:4), ], tmp.df)</code></pre>
<pre class="r"><code>base &lt;- expand.grid(unique(plot.df$cancer_type),
                    unique(plot.df$gene))
base$gene &lt;- rev(base$gene)
colnames(base) &lt;- c(&#39;cancer_type&#39;, &#39;gene&#39;)
base$cancer_type &lt;- factor(base$cancer_type,
                           levels = unique(cancer.levels))
base$gene &lt;- factor(base$gene,
                    levels = rev(unique(base$gene)))
genelevels &lt;- unique(base$gene[1:100]) %&gt;%
    str_replace_all(&quot;_&quot;, &quot; / &quot;) %&gt;%
    str_replace_all(&quot;KRAS / &quot;, &quot;&quot;)
base2 &lt;- base[c(1:100), ] %&gt;%
    mutate(gene=as.character(gene)) %&gt;%
    mutate(gene=str_replace_all(gene, &quot;_&quot;, &quot; / &quot;)) %&gt;%
    mutate(gene=str_replace_all(gene, &quot;KRAS / &quot;, &quot;&quot;)) %&gt;%
    mutate(gene=factor(gene, genelevels))
plot.df &lt;- plot.df %&gt;%
    mutate(gene=as.character(gene)) %&gt;%
    mutate(gene=str_replace_all(gene, &quot;_&quot;, &quot; / &quot;)) %&gt;%
    mutate(gene=str_replace_all(gene, &quot;KRAS / &quot;, &quot;&quot;)) %&gt;%
    mutate(gene=factor(gene, genelevels))
cm4 &lt;- ggplot(base2, aes(x = gene, y = cancer_type)) +
    geom_tile(fill = &#39;#ffffff&#39;,
              color = &#39;#deebf7&#39;,
              width = 0.8,
              height = 0.8) +
    theme_minimal() +
    geom_tile(data = plot.df %&gt;%
                  subset(gene %in% unique(base2[ , &#39;gene&#39;])),
              aes(fill = intersect.percent),
              size = 0.8,
              width = 0.8,
              height= 0.8) +
    scale_fill_distiller(palette = &#39;Reds&#39;,
                         name = &#39;Prevalence (%)&#39;,
                         values = rescale(1:10),
                         guide = guide_colorbar(direction = &quot;vertical&quot;),
                         trans = &quot;reverse&quot;) +
    theme(#axis.ticks.x = element_blank(),
        plot.margin = margin(-0.5,-0.5,-0.5,-0.5, &quot;cm&quot;),
        axis.text.x = element_text(size = 12,
                                   angle = 70,
                                   hjust=1),
        axis.text.y = element_text(size = 15, angle = 0),
        panel.grid = element_blank(),
        title = element_text(size = 8),
        strip.background = element_blank(),
        strip.text = element_blank(),
        legend.title=element_text(size=16),
        legend.text=element_text(size=14),
        plot.title=element_text(size=18))+
    ylab(&quot;&quot;) + xlab(&quot;&quot;) +
    ggtitle(&quot;KRAS G12C triple mutants&quot;)
aggregate.plot3 &lt;- plot_grid(NULL, cm4, NULL ,
                             nrow = 1,
                             rel_widths = c(0.25, 8, 0.25))
B &lt;- plot_grid(NULL, aggregate.plot3,
               NULL , ncol = 1,
               rel_heights = c(0.30, 8, 0.25))
Bg &lt;- ggplotGrob(B)
B</code></pre>
<p><img src="figure/fig5.Rmd/fig5b-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div
id="panel-c-concordance-of-estimated-gene-gene-dependencies-between-tmb-adjusted-model-and-multi-cancer-model"
class="section level1">
<h1>Panel C: Concordance of estimated gene-gene dependencies between TMB
adjusted model and multi-cancer model</h1>
<pre class="r"><code>#source(here(&quot;code&quot;, &quot;co-mutation&quot;, &quot;stan_funs.R&quot;))
##devtools::load_all(here(&quot;code&quot;, &quot;rasfuns&quot;))
cancerlabs &lt;- read_csv(here(&quot;data&quot;, &quot;cancer_labels.csv&quot;))</code></pre>
<pre><code>Rows: 51 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (2): cancer, cancer_label

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>colors &lt;- readRDS(here(&quot;output&quot;, &quot;co-mutation&quot;,
                       &quot;summarize_models.R&quot;,
                       &quot;colors.rds&quot;)) %&gt;%
    left_join(cancerlabs, by=&quot;cancer&quot;) %&gt;%
    mutate(cancer=cancer_label)
cols &lt;- setNames(colors$color, colors$cancer)</code></pre>
<pre class="r"><code>extdir &lt;- system.file(&quot;extdata&quot;, package=&quot;genie.6.1&quot;)
fname &lt;- file.path(extdir, &quot;tumor_normal_predictions.csv&quot;)
data(mutation_tmb, package=&quot;contingency.table&quot;)
data(patient_universe, package=&quot;genie.6.1&quot;)
matched_universe &lt;- filter(patient_universe, is_matched)
tmb_labels &lt;- read_csv(fname) %&gt;%
    rasfuns:::attach_tmb_label(matched_universe)</code></pre>
<pre><code>Rows: 39289 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (5): patient_id, sample_id, center, seq_assay_id, Estimate
dbl (3): Panel_Counts, Lower_Estimate, Upper_Estimate

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>Select the following: STK11 and KEAP1 in NSCLC TP53, inactivating APC
in CRC</p>
<pre class="r"><code>comutation &lt;- here(&quot;output&quot;, &quot;co-mutation&quot;, &quot;summarize_models.R&quot;,
                   &quot;combined.rds&quot;) %&gt;%
    readRDS() %&gt;%
    unnest(data) %&gt;%
    ungroup() %&gt;%
    left_join(cancerlabs, by=&quot;cancer&quot;) %&gt;%
    mutate(cancer=cancer_label) %&gt;%
    group_by(cancer_group, alteration_group, confounder) %&gt;%
    nest()

## significant associations in multi-cancer model
## -log10p &gt; 6
sig_multicancer &lt;- rasfuns:::significant_multicancer
filter_sig &lt;- rasfuns:::filter_by_signif
signif &lt;- sig_multicancer(comutation,
                          group_=&quot;Any mutation&quot;,
                          ras_=&quot;RAS_12-61&quot;,
                          p_cutoff=6)
isignif &lt;- sig_multicancer(comutation,
                           group_=&quot;Inactivating mutation&quot;,
                           ras_=&quot;RAS_12-61&quot;,
                           p_cutoff=6) %&gt;%
    filter(!nonras %in% signif$nonras &amp; cancer %in% signif$cancer)
signif &lt;- bind_rows(signif, isignif)
add_tmb_label &lt;- rasfuns:::add_tmb_label
filter_by &lt;- rasfuns:::filter_by
x  &lt;- comutation %&gt;%
    filter_by(c(&quot;Any mutation&quot;,
                &quot;Inactivating mutation&quot;),
              &quot;RAS_12-61&quot;, &quot;TMB&quot;) %&gt;%
    add_tmb_label(tmb_labels) %&gt;%
    filter_sig(signif)
ylist &lt;- rasfuns:::list_signif_by(signif, x$uid)
effect_list &lt;- rasfuns:::list_confounding(ylist, x)</code></pre>
<div id="co-mutation-of-ras-any-codon-and-any-non-ras-mutation"
class="section level2">
<h2>Co-mutation of RAS (any codon) and any non-RAS mutation</h2>
<p>Identify co-mutations in the stratified analysis with 95% credible
intervals that do not overlap the 95% credible interval in the
multi-cancer model. Focus on the set of co-mutations in which the
association in the multi-cancer model was statistically significant.</p>
<p>Several co-mutations significant in the multi-cancer model are not in
the stratified analyses. In the stratified analyses, we required that
the number of RAS mutations and the number of non-RAS mutations must
each be &gt; 100 (l. 133,
<code>contingency.table/data-derived/mutations_tmb.R</code>).</p>
<p>Below, we plot overall versus multi-cancer, highlighting the genes
with apparent discordance in 1 or more TMB strata.</p>
<pre class="r"><code>rename &lt;- dplyr::rename
multicancer &lt;- effect_list$data[[1]] %&gt;%
    select(nonras, alteration_group, cancer, `50%`) %&gt;%
    unite(&quot;uid&quot;, c(nonras, alteration_group, cancer)) %&gt;%
    rename(x=&quot;50%&quot;)
tmb  &lt;- effect_list$data[[2]] %&gt;%
    map_dfr(function(x) filter(x, grepl(&quot;mu_beta&quot;, Parameter))) %&gt;%
    select(nonras, alteration_group, cancer, `50%`, `2.5%`, `97.5%`) %&gt;%
    unite(&quot;uid&quot;, c(nonras, alteration_group, cancer)) %&gt;%
    rename(y=&quot;50%&quot;)
dat &lt;- left_join(multicancer, tmb, by=&quot;uid&quot;) %&gt;%
    separate(uid, c(&quot;nonras&quot;, &quot;alt_group&quot;, &quot;cancer&quot;), sep=&quot;_&quot;)
multicancer2 &lt;- effect_list$data[[3]] %&gt;%
    select(nonras, alteration_group, cancer, `50%`) %&gt;%
    unite(&quot;uid&quot;, c(nonras, alteration_group, cancer)) %&gt;%
    rename(x=&quot;50%&quot;)
tmb2  &lt;- effect_list$data[[4]] %&gt;%
    map_dfr(function(x) filter(x, grepl(&quot;mu_beta&quot;, Parameter))) %&gt;%
    select(nonras, alteration_group, cancer, `50%`, `25%`, `75%`) %&gt;%
    unite(&quot;uid&quot;, c(nonras, alteration_group, cancer)) %&gt;%
    rename(y=&quot;50%&quot;)
dat2 &lt;- left_join(multicancer2, tmb2, by=&quot;uid&quot;)  %&gt;%
    separate(uid, c(&quot;nonras&quot;, &quot;alt_group&quot;, &quot;cancer&quot;), sep=&quot;_&quot;)
all &lt;- bind_rows(dat, dat2) %&gt;%
    mutate(cancer=factor(cancer, sort(unique(cancer))))
dat &lt;- dat %&gt;%
    mutate(cancer=factor(cancer, levels(all$cancer)))
##stop()
cols &lt;- cols[levels(all$cancer)]
C &lt;- all %&gt;%
    ggplot(aes(x, y)) +
    geom_abline(intercept=0, slope=1) +
    geom_hline(yintercept=0, linetype=&quot;dashed&quot;, color=&quot;gray&quot;) +
    geom_vline(xintercept=0, linetype=&quot;dashed&quot;, color=&quot;gray&quot;) +
    geom_errorbar(aes(ymin=`25%`, ymax=`75%`),
                  color=&quot;gray&quot;, width=0.01) +
    geom_point(aes(color=cancer), size=point.size) +
    theme_bw(base_size=base.size) +
    ##geom_point(size=point.size, pch=21) +
    geom_point(data=dat,
               aes(color=cancer)) +
    geom_label_repel(data=dat,
                    aes(label=nonras),
                    fill=&quot;white&quot;, segment.color=&quot;black&quot;,
                    ##alpha=0.3,
                    size=gene.size,
                    max.overlaps=Inf,
                    box.padding=0.5) + ##color=&quot;transparent&quot;) +
    ylab(&quot;Cross-TMB average&quot;) +
    xlab(&quot;Association from multi-cancer model&quot;) +
    theme(panel.grid=element_blank(),
          legend.position=c(0.1, 0.8),
          ##legend.position=&quot;bottom&quot;,
          strip.background=element_blank(),
          strip.text=element_text(size=strip.size),
          axis.text=element_text(size=axis.text),
          axis.title=element_text(size=axis.title)) +
    xlim(c(-5, 3)) + ylim(c(-5, 3)) +
    guides(color=guide_legend(title=&quot;&quot;)) +
    scale_color_manual(values=cols)
##A_leg &lt;- cowplot::get_legend(fig)
##A &lt;- fig + guides(color=FALSE)
C</code></pre>
<p><img src="figure/fig5.Rmd/concordance-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Cg &lt;- ggplotGrob(C)</code></pre>
</div>
</div>
<div id="panel-d-gene-gene-dependencies-within-tmb-strata"
class="section level1">
<h1>Panel D: Gene-gene dependencies within TMB strata</h1>
<div
id="plot-intervals-for-overall-and-multicancer-for-genes-with-possible-tmb-confounding"
class="section level3">
<h3>plot intervals for overall and multicancer for genes with possible
tmb confounding</h3>
<pre class="r"><code>goi &lt;- c(&quot;STK11&quot;, &quot;KEAP1&quot;, &quot;TP53&quot;, &quot;APC&quot;)
multicancer &lt;- effect_list$data[[1]] %&gt;%
    select(nonras, alteration_group,
           cancer, n, `50%`, `2.5%`, `97.5%`) %&gt;%
    unite(&quot;uid&quot;, c(nonras, alteration_group, cancer), remove=FALSE) %&gt;%
    mutate(analysis=&quot;Multi-cancer&quot;) %&gt;%
    filter(nonras %in% goi)
tmb  &lt;- effect_list$data[[2]] %&gt;%
    map_dfr(function(x) filter(x, grepl(&quot;mu_beta&quot;, Parameter))) %&gt;%
    select(nonras, alteration_group,
           cancer, n, `50%`, `2.5%`, `97.5%`) %&gt;%
    unite(&quot;uid&quot;, c(nonras, alteration_group, cancer), remove=FALSE) %&gt;%
    filter(uid %in% multicancer$uid) %&gt;%
    mutate(analysis=&quot;Overall\n(TMB model)&quot;)
dat &lt;- bind_rows(multicancer, tmb) %&gt;%
    select(-uid) %&gt;%
    unite(uid, c(&quot;nonras&quot;, &quot;cancer&quot;), sep=&quot; in &quot;, remove=FALSE)
dat &lt;-  dat %&gt;%
    filter(!uid %in% &quot;TP53 in Pancreatic&quot;) %&gt;%
    mutate(analysis=factor(analysis,
                           c(&quot;Multi-cancer&quot;, &quot;Overall\n(TMB model)&quot;)),
           analysis=fct_rev(analysis))
tmb  &lt;- effect_list$data[[2]] %&gt;%
    map_dfr(function(x) filter(x, grepl(&quot;^beta&quot;, Parameter))) %&gt;%
    select(nonras, alteration_group,
           cancer, label, n, `50%`, `2.5%`, `97.5%`) %&gt;%
    unite(&quot;uid&quot;, c(nonras, alteration_group, cancer), remove=FALSE) %&gt;%
    filter(uid %in% multicancer$uid) %&gt;%
    select(-uid) %&gt;%
    unite(&quot;uid&quot;, c(nonras, cancer), sep=&quot; in &quot;, remove=FALSE)
levels &lt;- c(levels(dat$analysis), levels(tmb$label))
dat2 &lt;- dat %&gt;%
    rename(label=analysis)  %&gt;%
    bind_rows(tmb) %&gt;%
    mutate(label=factor(label, levels))

dat3 &lt;- dat2 %&gt;%
    filter(!uid %in% &quot;TP53 in Pancreatic&quot;) %&gt;%
    mutate(uid=factor(uid,
                      levels=c(&quot;KEAP1 in NSCLC&quot;,
                               &quot;STK11 in NSCLC&quot;, &quot;TP53 in NSCLC&quot;,
                               &quot;APC in CRC&quot;, &quot;TP53 in CRC&quot;)))
D &lt;- dat3 %&gt;%
    ggplot(aes(`50%`, label)) +
    geom_errorbarh(aes(xmin=`2.5%`, xmax=`97.5%`,
                       color=cancer), height=bar.height,
                   size=1.5) +
    geom_point(##pch=21, fill=&quot;white&quot;,
               size=point.size, aes(color=cancer)) +
    theme_bw(base_size=base.size) +
    geom_text(aes(x=Inf, y=label, label=n), hjust=1,
              size=n.size, color=&quot;gray30&quot;) +
    ylab(&quot;TMB strata&quot;) +
    xlab(&quot;Log odds ratio&quot;) +
    geom_hline(yintercept=2.5) +
    geom_vline(xintercept=0, linetype=&quot;dashed&quot;, color=&quot;gray&quot;) +
    theme(panel.border=element_rect(color=&quot;gray&quot;),
          axis.title=element_text(size=axis.title+1),
          ##axis.title.y=element_blank(),
          strip.text.y=element_blank(),
          strip.text.x=element_text(size=axis.title+2),
          axis.text.x=element_text(size=axis.text),
          axis.text.y=element_text(size=axis.text),
          panel.grid=element_blank(),
          legend.position=&quot;bottom&quot;,
          strip.background=element_blank()) +
    facet_wrap(~uid, ncol=5) +#, nrow=1) +
    scale_color_manual(values=cols) +
    guides(color=&#39;none&#39;)
D</code></pre>
<p><img src="figure/fig5.Rmd/paneld-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Dg &lt;- ggplotGrob(D)</code></pre>
</div>
</div>
<div id="figure-5" class="section level1">
<h1>Figure 5</h1>
<pre class="r"><code>grid.newpage()
gl &lt;- list(A,
           Bg,
           Cg,
           Dg,
           grob(),
           grob(),
           grob())
vp &lt;- viewport(width=unit(0.98, &quot;npc&quot;),
               height=unit(0.98, &quot;npc&quot;))
pushViewport(vp)
grid.arrange(grobs=gl,
             heights=c(1, 0.05, 1, 0.1, 1),
             widths=c(1, 0.1, 1),
             newpage=FALSE,
             layout_matrix=rbind(c(1, 6, 2),
                                 c(1, 6, 7), ## space between panels B and C
                                 c(1, 6, 3),
                                 c(5, 6, 5),
                                 c(4, 6, 4)))
upViewport()
grid.text(&quot;A&quot;, x=unit(0.015, &quot;npc&quot;), y=unit(0.98, &quot;npc&quot;),
          gp=gpar(cex=2.5))
grid.text(&quot;B&quot;, x=unit(0.5, &quot;npc&quot;), y=unit(0.98, &quot;npc&quot;),
          gp=gpar(cex=2.5))
grid.text(&quot;C&quot;, x=unit(0.5, &quot;npc&quot;), y=unit(0.67, &quot;npc&quot;),
          gp=gpar(cex=2.5))
grid.text(&quot;D&quot;, x=unit(0.015, &quot;npc&quot;), y=unit(0.33, &quot;npc&quot;),
          gp=gpar(cex=2.5))</code></pre>
<p><img src="figure/fig5.Rmd/fig5-1.png" width="2400" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
Ncells  4378782 233.9    7242198 386.8         NA  7242198 386.8
Vcells 17832970 136.1   42680404 325.7      16384 35497279 270.9</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin21.5.0 (64-bit)
Running under: macOS Monterey 12.5.1

Matrix products: default
BLAS:   /usr/local/Cellar/openblas/0.3.20/lib/libopenblasp-r0.3.20.dylib
LAPACK: /usr/local/Cellar/r/4.2.1/lib/R/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] rasfuns_1.1.5      ggrepel_0.9.1      here_1.0.1         scales_1.2.0      
 [5] readxl_1.4.0       RColorBrewer_1.1-3 gridExtra_2.3      forcats_0.5.1     
 [9] stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4        readr_2.1.2       
[13] tidyr_1.2.0        tibble_3.1.8       tidyverse_1.3.2    cowplot_1.1.1     
[17] reshape2_1.4.4     plyr_1.8.7         ggplot2_3.3.6      workflowr_1.7.0   

loaded via a namespace (and not attached):
 [1] matrixStats_0.62.0     bitops_1.0-7           fs_1.5.2              
 [4] bit64_4.0.5            lubridate_1.8.0        httr_1.4.3            
 [7] rprojroot_2.0.3        GenomeInfoDb_1.32.3    tools_4.2.1           
[10] backports_1.4.1        bslib_0.4.0            utf8_1.2.2            
[13] R6_2.5.1               DBI_1.1.3              BiocGenerics_0.42.0   
[16] colorspace_2.0-3       withr_2.5.0            tidyselect_1.1.2      
[19] processx_3.7.0         bit_4.0.4              compiler_4.2.1        
[22] git2r_0.30.1           cli_3.3.0              rvest_1.0.2           
[25] xml2_1.3.3             labeling_0.4.2         sass_0.4.2            
[28] callr_3.7.1            digest_0.6.29          rmarkdown_2.14        
[31] XVector_0.36.0         pkgconfig_2.0.3        htmltools_0.5.3       
[34] highr_0.9              dbplyr_2.2.1           fastmap_1.1.0         
[37] rlang_1.0.4            rstudioapi_0.13        farver_2.1.1          
[40] jquerylib_0.1.4        generics_0.1.3         jsonlite_1.8.0        
[43] vroom_1.5.7            googlesheets4_1.0.1    RCurl_1.98-1.8        
[46] magrittr_2.0.3         GenomeInfoDbData_1.2.8 S4Vectors_0.34.0      
[49] Rcpp_1.0.9             munsell_0.5.0          fansi_1.0.3           
[52] lifecycle_1.0.1        stringi_1.7.8          whisker_0.4           
[55] yaml_2.3.5             zlibbioc_1.42.0        parallel_4.2.1        
[58] promises_1.2.0.1       crayon_1.5.1           Biostrings_2.64.0     
[61] haven_2.5.0            hms_1.1.1              knitr_1.39            
[64] ps_1.7.1               pillar_1.8.0           stats4_4.2.1          
[67] reprex_2.0.1           glue_1.6.2             evaluate_0.16         
[70] getPass_0.2-2          modelr_0.1.8           EnhancedVolcano_1.14.0
[73] vctrs_0.4.1            tzdb_0.3.0             httpuv_1.6.5          
[76] cellranger_1.1.0       gtable_0.3.0           assertthat_0.2.1      
[79] cachem_1.0.6           xfun_0.32              broom_1.0.0           
[82] later_1.3.0            googledrive_2.0.0      gargle_1.2.0          
[85] IRanges_2.30.0         ellipsis_0.3.2        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
