---
title: "Figure S18: KRAS mutant allele and overall survival -- TCGA cohort"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r packages, include=FALSE}
library(grid)
library(ggplot2)
library(survival)
library(survminer)
library(cowplot)
library(tidyverse)
library(gridExtra)
library(here)
```

```{r}
clinical.df  <- readRDS(here("data", "tcga_clinical_comut.rds")) %>%
    as_tibble()
## Not clear how genes were selected
genes <- c("KEAP1", "NTRK3", "PIK3CA", "TP53", "RBM10", "STK11")
ras <- c(rep("KRAS_G12C", 4), rep("KRAS_G12V", 2))
analyses <- paste(ras, genes, sep="~")
clin.df <- clinical.df %>%
    rename(time=os,
           status=os.cens) %>%
    filter(type=="LUAD") %>%
    select(time, status, all_of(analyses))
```

```{r plot_survival}
plot_survival <- function(dat, x.pval, y.pval){
    diff.fit <- survdiff(Surv(time, status) ~ varlabel, data = dat)
    logrank.pvalue = formatC(1 - pchisq(diff.fit$chisq,
                                        length(diff.fit$n) - 1),
                             digits = 2)
    pval <- paste("Log rank p =", logrank.pvalue)
    cox.survfit.km <- survfit(Surv(time, status) ~ varlabel,
                              data = dat)
    fig <- ggsurvplot(cox.survfit.km, data = dat,
                     risk.table = TRUE,
                     tables.height = 0.2,
                     fontsize=12, tables.y.text = FALSE,
                     tables.theme = theme_cleantable(),
                     risk.table.fontsize = 4.0,
                     risk.table.y.text.col = T,
                     risk.table.y.text = FALSE,
                     font.legend = c(12, "plain", "black"),
                     xlab="Months",
                     fontsize.title = 1,
                     risk.table.col = "strata",
                     risk.table.title=NULL,
                     font.title = c(12, "plain", "black"),
                     conf.int=FALSE,
                     break.time.by = 10,
                     font.x = c(12, "plain", "black"),
                     font.y = c(12, "plain", "black"),
                     font.xtickslab = c(12, "plain", "black"),
                     font.ytickslab = c(12, "plain", "black"))
    gfig <- fig$plot +
        scale_color_manual(values=c("#377EB8", "#E41A1C"),
                           labels=unique(dat$varlabel),
                           name="") +
        theme(legend.direction="vertical",
              legend.position=c(0.6, 0.9),
              axis.title.x=element_text(size=15),
              axis.text.x=element_text(angle=90),
              axis.title.y=element_text(size=15),
              plot.title=element_text(hjust=0.5, size=17)) +
        annotate("text", x=x.pval, y=y.pval, label=pval, hjust=0) +
        ylab("Overall Survival (Prob)") +
        ggtitle(unique(dat$title))
    g1 <- ggplotGrob(gfig)
    gtab <- fig$table +
        guides(color="none")
    g2 <- ggplotGrob(gtab)
    g2$widths <- g1$widths
    glist <- arrangeGrob(grobs=list(g1,
                                    g2),
                         heights=c(0.75, 0.25),
                         layout_matrix=matrix(1:2, ncol=1))
    glist
}
```

```{r surv_grob}
clin.df2 <- clin.df %>%
    pivot_longer(cols=starts_with("KRAS"),
                 names_to="variant",
                 values_to="is_altered") %>%
    mutate(varlabel=ifelse(is_altered==1, variant, "WT"),
           varlabel=str_replace_all(varlabel, "_", " "),
           varlabel=str_replace_all(varlabel, "~", " / ")) %>%
    group_by(variant) %>%
    nest() %>%
    mutate(title=paste("LUAD", variant),
           title=str_replace_all(title, "_", " "),
           title=str_replace_all(title, "~", " / ")) %>%
    mutate(title=factor(title, title))  %>%
    unnest("data") %>%
    group_by(variant) %>%
    nest()
varlabel_as_factor <- function(x){
    x$varlabel <- factor(x$varlabel, unique(x$varlabel))
    x
}
clin.df2$data <- clin.df2$data %>%
    map(varlabel_as_factor)
glist <- clin.df2$data %>%
    map(plot_survival, x.pval=5, y.pval=0.1)
```

```{r fig_s18, fig.width=22, fig.height=18, dev=c("png", "pdf")}
##glist[[3]] <- nullGrob()
glist[[7]] <- nullGrob()
grid.newpage()
vp <- viewport(width=unit(0.96, "npc"), height=unit(0.96, "npc"))
pushViewport(vp)
mat <- cbind(c(1, 3, 5), 7, c(2, 4, 6))
grid.arrange(grobs=glist,
             widths=c(1, 0.1,  1),
             heights=c(1, 1, 1),
             layout_matrix=mat,
             newpage=FALSE)
upViewport()
grid.text(LETTERS[1:2],
          x=unit(c(0.015, .51), "npc"),
          y=unit(rep(.975, 2), "npc"),
          gp=gpar(cex=2))
grid.text(LETTERS[3:4],
          x=unit(c(0.015, .51), "npc"),
          y=unit(rep(.66, 2), "npc"),
          gp=gpar(cex=2))
grid.text(LETTERS[5:6],
          x=unit(c(0.015, .51), "npc"),
          y=unit(rep(.33, 2), "npc"),
          gp=gpar(cex=2))
```
