---
title: "Table S3:  Co-occurrence and mutual exlusivity of KRAS variants at codons 12,13, and 61 with non-RAS gene mutations computed by the DISCOVER and SELECT methods"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

```{r packages, message=FALSE, warning=FALSE, echo=FALSE}
knitr::opts_chunk$set(autodep = TRUE, echo=FALSE)
library(magrittr)
library(tidyverse)
library(gt)
library(here)
library(DT)
```

```{r data, include=FALSE, message=FALSE, echo=FALSE}
discover.dat <- here("output", "methods-comparison",
                     "genie_nsclc_discover.tsv")%>%
            read_tsv()  %>%
              dplyr::select_all(~paste0("discover.", .)) %>%
            mutate(method = "DISCOVER") 

discover.dat <- discover.dat[c(grep("KRAS_12-61", discover.dat$`discover.gene1` ), grep("KRAS_12-61", discover.dat$`discover.gene2` )),]


select.dat <- here("output", "methods-comparison",
                     "genie_nsclc_select.tsv")%>%
            read_tsv() %>%
            dplyr::select_all(~paste0("select.", .)) %>%
            mutate(method = "SELECT")

select.dat <- select.dat[c(grep("KRAS_12-61", select.dat$`select.SFE_1` ), grep("KRAS_12-61", select.dat$`select.SFE_2` )),]

```

```{r table_s3}
capitalize <- Hmisc::capitalize
tab <-   plyr::rbind.fill(discover.dat, select.dat)

tab %>%
    datatable(extensions = 'Buttons',
              options = list(dom = 'Blfrtip',
                             buttons = c('copy', 'csv', 'excel',
                                         'pdf', 'print'),
                             lengthMenu = list(c(50,100,-1),
                                               c(50,100,"All"))))

```
